src/streamdiffusion/stream_parameter_updater.py

```diff
@@ -1206,65 +1206,69 @@ def _update_ipadapter_config(self, desired_config: Dict[str, Any]) -> None:
-                    raise
+                            self.stream.ipadapter.set_scale(desired_scale)
+                        # Update our tracking attribute
+                        setattr(self.stream.ipadapter, 'scale', desired_scale)
+                    except Exception:
+                        # Do not add fallback mechanisms
+                        raise
+                else:
+                    # Simple uniform scale
+                    if hasattr(self.stream, 'ipadapter'):
+                        self.stream.ipadapter.set_scale(desired_scale)
+                        setattr(self.stream.ipadapter, 'scale', desired_scale)
         
 
         # Update weight type if provided (affects per-layer distribution and/or per-step factor)
-        if 'weight_type' in desired_config:
+        if 'weight_type' in desired_config and desired_config['weight_type'] is not None:
             weight_type = desired_config['weight_type']
-            try:
-                setattr(self.stream, 'ipadapter_weight_type', weight_type)
-            except Exception:
-                pass
-            # For PyTorch UNet, immediately apply a per-layer scale vector so layers reflect selection types
-            try:
-                is_tensorrt_engine = hasattr(self.stream.unet, 'engine') and hasattr(self.stream.unet, 'stream')
-                if not is_tensorrt_engine and hasattr(ipadapter_pipeline, 'ipadapter') and ipadapter_pipeline.ipadapter is not None:
-                    # Compute per-layer vector using Diffusers_IPAdapter helper
-                    from diffusers_ipadapter.ip_adapter.attention_processor import build_layer_weights
-                    # Count installed IP layers by scanning processors with _ip_layer_index
-                    ip_procs = [p for p in self.stream.pipe.unet.attn_processors.values() if hasattr(p, "_ip_layer_index")]
-                    num_layers = len(ip_procs)
-                    base_weight = float(getattr(self.stream, 'ipadapter_scale', getattr(ipadapter_pipeline, 'scale', 1.0)))
-                    weights = build_layer_weights(num_layers, base_weight, weight_type)
-                    # If None, keep uniform base scale; else set per-layer vector
-                    if weights is not None:
-                        ipadapter_pipeline.ipadapter.set_scale(weights)
-                    else:
-                        ipadapter_pipeline.ipadapter.set_scale(base_weight)
-            except Exception:
-                # Do not add fallback mechanisms
-                raise
+            # Update IPAdapter instance
+            if hasattr(self.stream, 'ipadapter'):
+                setattr(self.stream.ipadapter, 'weight_type', weight_type)
+                
+                # For PyTorch UNet, immediately apply a per-layer scale vector so layers reflect selection types
+                try:
+                    is_tensorrt_engine = hasattr(self.stream.unet, 'engine') and hasattr(self.stream.unet, 'stream')
```

Comment:
This is a more general comment. Not a comment for this PR specifically nor sth I think you should address here, but it's a general practice I've seen and been confused by on the little bit I poked around in this repo:

I feel like this practice of relying on attributes existing or not and have conditionals and defaults repeated everywhere leads to much unclearer code. They could almost always be replaced with proper field definitions with sane defaults (even if `None`, which is still cleaner to check than `has/get/setattr`). It's also harder to have proper typing if we don't declare the fields in a single place, and especially if we change field values from outside the object without any setter,

WDYT?